sklearn/gaussian_process/_gpr.py

```diff
@@ -48,16 +48,11 @@ class GaussianProcessRegressor(MultiOutputMixin,
         passed, the kernel "1.0 * RBF(1.0)" is used as default. Note that
         the kernel's hyperparameters are optimized during fitting.
 
-    alpha : float or array-like of shape (n_samples), default=1e-10
-        Value added to the diagonal of the kernel matrix during fitting.
-        Larger values correspond to increased noise level in the observations.
-        This can also prevent a potential numerical issue during fitting, by
-        ensuring that the calculated values form a positive definite matrix.
-        If an array is passed, it must have the same number of entries as the
-        data used for fitting and is used as datapoint-dependent noise level.
-        Note that this is equivalent to adding a WhiteKernel with c=alpha.
-        Allowing to specify the noise level directly as a parameter is mainly
-        for convenience and for consistency with Ridge.
+    alpha : float, optional (default: 1e-10)
```

Comment:
```suggestion
    alpha : float, default=1e-10
```

Reply:
```suggestion
    alpha : float or ndarray of shape (n_samples,), default=1e-10
```
At least, an ndarray is still a valid option. Any reason to not document it anymore?

Reply:
Yeah fair enough. I think there is a reason but it'll be better to raise as a separate issue at some point otherwise this could get very messy!

Reply:
It's a valid option. If you adjust this docstring anyway, it would be nice to do it right at once.
I could imagine a separate PR for testing ndarray alpha as I couldn't find any test!